src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,299 @@
+    // is defined by the sampled theta, and the muon neutrino direction is
+    // calculated via conservation of momentum
+    Real3 charged_lep_dir = {0, 0, 1};
+    Real3 electron_nu_dir = {sin_theta, 0, cos_theta};
+    Real3 muon_nu_dir
+        = {-(electron_nu_energy_frac / muon_nu_energy_frac) * sin_theta,
+           0,
+           -electron_energy_frac / muon_nu_energy_frac
+               - (electron_nu_energy_frac / muon_nu_energy_frac) * cos_theta};
+
+    // Apply a spherically uniform rotation to the decay
+    real_type euler_phi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
+    real_type euler_theta = std::acos(2 * generate_canonical(rng) - 1);
+    real_type euler_psi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
+
+    EulerRotation rotate(euler_phi, euler_theta, euler_psi);
+    charged_lep_dir = rotate(charged_lep_dir);
+    electron_nu_dir = rotate(electron_nu_dir);
+    muon_nu_dir = rotate(muon_nu_dir);
+
+    // Boost secondaries to the lab frame
+    auto charged_lep_4vec = this->to_lab_frame(
+        charged_lep_dir, charged_lep_energy, shared_.electron_mass);
+    auto electron_nu_4vec
+        = this->to_lab_frame(electron_nu_dir, electron_nu_energy, 0);
+    auto muon_nu_4vec = this->to_lab_frame(muon_nu_dir, muon_nu_energy, 0);
+
+    // \todo Return electron only?
+    Interaction result = Interaction::from_absorption();
+    result.action = Interaction::Action::decay;
+    result.secondaries = {secondaries, 3};
+
+    // Interaction stores kinetic energy; FourVector stores total energy
+    result.secondaries[0].particle_id = secondary_id_[0];
+    result.secondaries[0].energy
+        = Energy{charged_lep_4vec.energy - shared_.electron_mass};
+    result.secondaries[0].direction = make_unit_vector(charged_lep_4vec.mom);
+
+    result.secondaries[1].particle_id = secondary_id_[1];
+    result.secondaries[1].energy = Energy{electron_nu_4vec.energy};  // no mass
+    result.secondaries[1].direction = make_unit_vector(electron_nu_4vec.mom);
+
+    result.secondaries[2].particle_id = secondary_id_[2];
+    result.secondaries[2].energy = Energy{muon_nu_4vec.energy};  // no mass
+    result.secondaries[2].direction = make_unit_vector(muon_nu_4vec.mom);
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Boost secondary to the lab frame.
+ *
+ * \note This assumes the primary to be at rest and, thus, there is no need to
+ * perform an inverse boost of the primary at the CM frame.
+ */
+CELER_FUNCTION FourVector MuDecayInteractor::to_lab_frame(Real3 const& dir,
+                                                          Energy const& energy,
+                                                          real_type const& mass)
+{
+    CELER_EXPECT(norm(dir) > 0);
+    CELER_EXPECT(energy > zero_quantity());
+    CELER_EXPECT(mass >= 0);
+
+    Real3 p = dir * energy.value();  // Momentum [MeV]
```

Comment:
??

Reply:
Equivalent to `inc_direction_ * particle.momentum().value()`, since [E] = [p].